build/linux/installer/scripts/tomlparser-agent-config.rb

```diff
@@ -131,6 +137,40 @@ def populateSettingValuesFromConfigMap(parsedConfig)
           puts "Using config map value: NODES_EMIT_STREAM_BATCH_SIZE = #{@nodesEmitStreamBatchSize}"
         end
       end
+      # fbit config settings
+      fbit_config = parsedConfig[:agent_settings][:fbit_config]
+      if !fbit_config.nil?
+        fbitFlushInterval = fbit_config[:FBIT_SERVICE_FLUSH_INTERVAL]
+        if !fbitFlushInterval.nil? && is_number?(fbitFlushInterval) && fbitFlushInterval.to_i > 0
+          @fbitFlushInterval = fbitFlushInterval.to_i
+          puts "Using config map value: FBIT_SERVICE_FLUSH_INTERVAL = #{@fbitFlushInterval}"
+        end
+
+        fbitTailBufferChunkSize = fbit_config[:FBIT_TAIL_BUFFER_CHUNK_SIZE]
+        if !fbitTailBufferChunkSize.nil? && is_number?(fbitTailBufferChunkSize) && fbitTailBufferChunkSize.to_i > 0
+          @fbitTailBufferChunkSize = fbitTailBufferChunkSize.to_i
+          puts "Using config map value: FBIT_TAIL_BUFFER_CHUNK_SIZE = #{@fbitTailBufferChunkSize}"
+        end
+
+        fbitTailBufferMaxSize = fbit_config[:FBIT_TAIL_BUFFER_MAX_SIZE]
+        if !fbitTailBufferMaxSize.nil? && is_number?(fbitTailBufferMaxSize) && fbitTailBufferMaxSize.to_i > 0 
+          if fbitTailBufferMaxSize.to_i >= @fbitTailBufferChunkSize
```

Comment:
can you explain this optimization and where it says it cannot be > ?

Reply:
> can you explain this optimization and where it says it cannot be > ?

Look at these,
https://fluentbit.io/documentation/0.13/input/tail.html, default value of Buffer_Max_Size is the same as Buffer_Chunk_Size. 
https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_config.c#L165

Reply:
if its not specified its the same. where oes it say it cannot be > ?

Reply:
> if its not specified its the same. where does it say it cannot be > ?

> here https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_config.c#L165.

Reply:
just verified. if this is not specified td-agent crashing so we need to handle this case to avoid case in case if user provided one value but not other.